src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import { FormEvent, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const findUserById = (userId: number) => {
+  return usersFromServer.find((user) => (
+    user.id === userId
+  )) || null;
+};
+
+const todos = todosFromServer.map((todo) => {
+  return {
+    ...todo,
+    user: findUserById(todo.userId),
+  };
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState('');
+  const [visibleTodos, setTodos] = useState(todos);
+  const [titleValid, checkTitleValid] = useState(true);
```

Comment:
if you store a `boolean` value - just start name with `is`, `has`
```suggestion
  const [isTitleValid, setIsTitleValid] = useState(true);
```